src/utils.js

```diff
@@ -0,0 +1,53 @@
+import { getLayer } from 'ol-mapbox-style';
+import { Point } from 'ol/geom.js';
+import RenderFeature from 'ol/render/Feature.js';
+
+/**
+ * gets the ol layer for a given mapbox source-layer
+ * @param {string} mapboxLayerId mapbox source layer id
+ * @returns {import("ol/layer/Vector.js").default | import("ol/layer/VectorTile.js").default}
+ */
+export function getOlLayerForMbLayer(contentLayer, mapboxLayerId) {
+  if (!mapboxLayerId) {
+    return;
+  }
+  const styleDocument = contentLayer.get('mapbox-style');
+  if (!styleDocument) {
+    return;
+  }
+
+  const mapboxLayer = styleDocument.layers.find(l => {
+    return l['source-layer'] === mapboxLayerId;
+  });
+
+  if (!mapboxLayer) {
+    throw new Error(
+      `<yio-map>: Invalid value for "editLayer". Could not find "${mapboxLayerId}" as source-layer in the content map.`,
+    );
+  }
+  return /** @type {import("ol/layer/Vector.js").default | import("ol/layer/VectorTile.js").default} */ (
+    getLayer(contentLayer, mapboxLayer.id)
+  );
+}
+
+/**
+ * gets the ID of a feature or render feature
+ * @param {import("ol/Feature.js").FeatureLike} featureOrRenderFeature
+ */
+export function getFeatureId(featureOrRenderFeature) {
+  if (featureOrRenderFeature instanceof RenderFeature) {
+    return featureOrRenderFeature.get('id');
+  }
+  return featureOrRenderFeature.getId();
+}
+
+/**
+ * @param {import("ol/Feature.js").FeatureLike} featureOrRenderFeature
+ * @returns {boolean} true if geometry is a point
+ */
+export function isPointFeature(featureOrRenderFeature) {
+  if (featureOrRenderFeature instanceof RenderFeature) {
+    return featureOrRenderFeature.getType() === 'Point';
+  }
+  return featureOrRenderFeature.getGeometry() instanceof Point;
+}
```

Comment:
Regardless of whether the feature is a `Feature` or a `RenderFeature`, you can always use
```js
feature.getGeometry().getType() === 'Point';
```